### PR TITLE
Improve transport

### PR DESCRIPTION
**Pull request by tommy5oo**
Improved channels in transport
Add unauthenticated error handling
Remove kotlin-reflect
Remove unused code

### CODE REVIEW

File: nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/core/RegistryService.kt [hunk 1]
```diff
@@ -43,7 +43,7 @@ internal class RegistryService(private val observer: GrpcServiceListener) :
             if(blockingStub == null || channel == null) {
                 return false
             }
-            val timeout = channel!!.opts.connectionTimeout.toLong()
+            val timeout = channel!!.defaultOptions.connectionTimeout.toLong()
```
Comment (shshshl09) [inline nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/core/RegistryService.kt]: Should be replace defalutOptions to currentOptions?

Comment (tommy5oo) [reply nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/core/RegistryService.kt]: done! :)

Comment (tommy5oo) [reply nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/core/RegistryService.kt]: > Should be replace defalutOptions to currentOptions?

done! :)

File: nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/GrpcTransport.kt [hunk 2]
```diff
@@ -409,8 +397,10 @@ internal class GrpcTransport(
                 }
             }
             AuthStateListener.State.REFRESHED -> {
-                if (State.AUTHORIZING == state) {
-                    setState(State.CONNECTING)
+                when (state) {
+                    State.CONNECTED -> setState(State.DISCONNECTING)
```
Comment (shshshl09) [inline nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/GrpcTransport.kt]: missing state update?

File: nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/GrpcTransport.kt [hunk 3]
```diff
@@ -534,4 +534,9 @@ internal class GrpcTransport(
 
         connect()
     }
+
+    override fun onUnAuthenticated() {
+        setState(State.WAITING_TO_RETRY_CONNECTING)
```
Comment (shshshl09) [inline nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/GrpcTransport.kt]: I think that the state should be disconnected

File: nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/GrpcTransport.kt (hunk 2 above)
Comment (tommy5oo) [reply nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/GrpcTransport.kt]: right

File: nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/GrpcTransport.kt (hunk 3 above)
Comment (tommy5oo) [reply nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/port/transport/grpc/GrpcTransport.kt]: WAITING_TO_RETRY_CONNECTING is correct

**Pull request merged**